lib/CodeBuilder/Adapter/TolerantParser/Updater/ClassMethodUpdater.php

```diff
@@ -2,16 +2,20 @@
 
 namespace Phpactor\CodeBuilder\Adapter\TolerantParser\Updater;
 
+use Microsoft\PhpParser\Node\ClassMembersNode;
 use Microsoft\PhpParser\ClassLike;
+use Microsoft\PhpParser\Node\InterfaceMembers;
 use Microsoft\PhpParser\Node\Statement\ClassDeclaration;
 use Microsoft\PhpParser\Node\Statement\TraitDeclaration;
+use Microsoft\PhpParser\Node\TraitMembers;
 use Phpactor\CodeBuilder\Domain\Renderer;
 use Phpactor\CodeBuilder\Domain\Prototype\Method;
 use RuntimeException;
+use Microsoft\PhpParser\Node;
 
 class ClassMethodUpdater extends AbstractMethodUpdater
 {
-    public function memberDeclarationsNode(ClassLike $classNode)
+    public function memberDeclarationsNode(ClassLike $classNode): ClassMembersNode|TraitMembers|InterfaceMembers
```

Comment:
what about EnumMembers?

Reply:
but this cannot return anything other then ClassMembersNode? same for the other "updaters" I guess, they should only return the corresponding `*MembersNode`

Reply:
Correct EnumMembers are missing. I will add them but we don't have an enum updater that's why I forgot / left it out.

Reply:
we should use a template type here.

```
/**
 * @template T
 * @extends AbstractMethodUpdate<ClassMembersNode>
 */
class ClassMethodUpdater ...

/**
 * @template TMembersNodeType of Node
 */
class AbstractMethodUpdater ...
{
   /** @return TMembersNodeType */
    abstract public function memberDeclrationsNode(...)
}
```